spec/lib/hackney/income/migrate_uh_agreement_spec.rb

```diff
@@ -0,0 +1,195 @@
+
+    before do
+      expect(view_agreements).to receive(:execute).and_return([new_agreement])
+    end
+
+    it 'nothing get\'s migrated' do
+      expect(view_court_cases).not_to receive(:execute)
+      expect(create_informal_agreement).not_to receive(:execute)
+      expect(create_formal_agreement).not_to receive(:execute)
+      expect(create_agreement_migration).not_to receive(:execute)
+
+      subject.migrate(tenancy_ref: tenancy_ref)
+    end
+  end
+
+  context 'when there is a UH agreement but no MA agreements' do
+    let(:new_agreement) { create(:agreement) }
+
+    let(:amount) { 3.55 }
+    let(:uh_id) { 110_639_718 }
+    let(:starting_balance) { 273.52 }
+
+    let(:uh_agreements) {
+      [{
+        start_date: '2012-12-24',
+        status: '400       ',
+        breached: true,
+        last_check_balance: 154.0,
+        last_check_date: '2013-11-30',
+        last_check_expected_balance: 99.57,
+        starting_balance: starting_balance,
+        comment: ' ',
+        amount: amount,
+        frequency: 1,
+        uh_id: uh_id
+      }]
+    }
+
+    before do
+      expect(view_agreements).to receive(:execute).and_return([])
+      expect(view_court_cases).to receive(:execute).and_return([])
+    end
+
+    it 'migrates an informal agreement' do
+      expect(create_informal_agreement).to receive(:execute).with(
+        new_agreement_params: {
+          agreement_type: :informal,
+          amount: amount,
+          court_case_id: nil,
+          created_by: 'Managed Arrears migration from UH',
+          frequency: 0,
+          notes: ' ',
+          start_date: '2012-12-24',
+          starting_balance: starting_balance,
+          tenancy_ref: tenancy_ref
+        }
+      ).and_return(new_agreement)
+
+      expect(create_formal_agreement).not_to receive(:execute)
+
+      expect(create_agreement_migration).to receive(:execute).with(
+        agreement_migration_params: {
+          agreement_id: new_agreement.id,
+          legacy_id: uh_id
+        }
+      )
+
+      subject.migrate(tenancy_ref: tenancy_ref)
+    end
+  end
+
+  context 'when there is a UH agreement, no MA agreements and a court case' do
+    let(:new_agreement) { create(:agreement) }
+    let(:court_case) { create(:court_case) }
+
+    let(:amount) { 3.55 }
+    let(:uh_id) { 110_639_718 }
+    let(:starting_balance) { 273.52 }
+
+    let(:uh_agreements) {
+      [{
+        start_date: '2012-12-24',
+        status: '400       ',
+        breached: true,
+        last_check_balance: 154.0,
+        last_check_date: '2013-11-30',
+        last_check_expected_balance: 99.57,
+        starting_balance: starting_balance,
+        comment: ' ',
+        amount: amount,
+        frequency: 4,
+        uh_id: uh_id
+      }]
+    }
+
+    before do
+      expect(view_agreements).to receive(:execute).and_return([])
+      expect(view_court_cases).to receive(:execute).and_return([court_case])
+    end
+
+    it 'migrates an informal agreement' do
```

Comment:
```suggestion
    it 'migrates a formal agreement' do
```
bit nitpicky 😅  but it's this right? 🤔